src/Init/Data/String/Basic.lean

```diff
@@ -490,6 +490,11 @@ if s.isNat then
 else
   none
 
+def beq (ss1 ss2 : Substring) : Bool :=
+ss1.toString == ss2.toString
```

Comment:
This could be sped up in a native implementation. Right now I only used it in tests, via `Syntax.structEq`.